packages/hawtio/src/plugins/camel/camel-preferences-service.ts

```diff
@@ -0,0 +1,95 @@
+export interface ICamelStorage {
+  loadIsHideOptionDocumentation(): boolean
+  loadIsHideDefaultOptionValues(): boolean
+  loadIsHideUnusedOptionValues(): boolean
+  loadIsIncludeTraceDebugStreams(): boolean
+  loadMaximumTraceDebugBodyLength(): number
+  loadMaximumLabelWidth(): number
+  loadIsIgnoreIDForLabel(): boolean
+  loadIsShowInflightCounter(): boolean
+  loadRouteMetricMaximumSeconds(): number
+  loadCamelPreferences(): ICamelPreferences
+  saveCamelPreferences(newValues: Partial<ICamelPreferences>): void
+}
+
+export interface ICamelPreferences {
+  isHideOptionDocumentation: boolean
+  isHideDefaultOptionValues: boolean
+  isHideUnusedOptionValues: boolean
+  isIncludeTraceDebugStreams: boolean
+  maximumTraceDebugBodyLength: number
+  maximumLabelWidth: number
+  isIgnoreIDForLabel: boolean
+  isShowInflightCounter: boolean
+  routeMetricMaximumSeconds: number
+}
+
+const CAMEL_PREFERENCES_DEFAULT_VALUES: ICamelPreferences = {
+  isHideOptionDocumentation: false,
+  isHideDefaultOptionValues: false,
+  isHideUnusedOptionValues: false,
+  isIncludeTraceDebugStreams: false,
+  maximumTraceDebugBodyLength: 5000,
+  maximumLabelWidth: 34,
+  isIgnoreIDForLabel: false,
+  isShowInflightCounter: true,
+  routeMetricMaximumSeconds: 10,
+}
+
+export const STORAGE_KEY_CAMEL_PREFERENCES = 'camel.preferences'
+
+class CamelPreferencesService implements ICamelStorage {
+  loadCamelPreferences(): ICamelPreferences {
+    return { ...CAMEL_PREFERENCES_DEFAULT_VALUES, ...this.loadFromStorage() }
+  }
+
+  saveCamelPreferences(newValues: Partial<ICamelPreferences>): void {
```

Comment:
Deconstruction + Partial type makes for a very easy way to ensure that we don't erase any unset keys.